lib/facter/simplib__numa.rb

```diff
@@ -0,0 +1,78 @@
+# _Description_
+#
+# Return true if system has multiple NUMA nodes
+#
+Facter.add("simplib__numa_enabled") do
+  confine :kernel => 'Linux'
+  confine { File.exist? '/sys/devices/system/node' }
+
+  setcode do
+    true
+  end
+
+end
+
+# _Description_
+#
+# Return supported NUMA possible nodes
+#
+Facter.add("simplib__numa_possible_nodes") do
+  confine :kernel => 'Linux'
+  confine { File.exist? '/sys/devices/system/node/possible' }
+
+  # read the file
+  File.open('/sys/devices/system/node/possible', 'r') do | text |
+    result = text.read()
+  end
+
+  setcode do
+    result
+  end
+
+end
+
+# _Description_
+#
+# Return supported NUMA online nodes
+#
+Facter.add("simplib__numa_online_nodes") do
+  confine :kernel => 'Linux'
+  confine { File.exist? '/sys/devices/system/node/online' }
+
+  # read the file
+  File.open('/sys/devices/system/node/online', 'r') do | text |
+    result = text.read()
+  end
+
+  setcode do
```

Comment:
Same `File.read()` change here and toss everything that's not a `confine` into the `setcode` for isolation.